frontend/src/components/pages/AdminDashboard/ViewDonationsAndCheckIns.tsx

```diff
@@ -52,6 +86,25 @@ const ViewDonationsAndCheckIns = ({
     getCheckIns();
   }, []);
 
+  // filter donations/schedules based on selected filter
+  useEffect(() => {
+    let filteredDonations;
+
+    if (selectedFilter === DonationFilterType.ALL)
+      filteredDonations = schedules;
+    else if (selectedFilter === DonationFilterType.FILLED)
+      filteredDonations = schedules.filter(
+        (schedule) => schedule.volunteerId != null,
+      );
+    // selectedFilter === AccountFilterType.DONOR
+    else
+      filteredDonations = schedules.filter(
+        (schedule) => schedule.volunteerId == null,
+      );
+
+    setFilteredSchedules(filteredDonations);
+  }, [selectedFilter]);

```

Comment:
can we memoize this?
```suggestion
  React.useMemo(()=>{
    if (selectedFilter === DonationFilterType.FILLED){
      setFilteredSchedules([...schedules.filter((schedule) => schedule.volunteerId != null)]);
    }
    else if (selectedFilter === DonationFilterType.UNASSIGNED){
      setFilteredSchedules([...schedules.filter((schedule) => schedule.volunteerId == null)]);
    }
    else {
      setFilteredSchedules([...schedules]);
    }
  }, [selectedFilter]);
```